Generator.php

```diff
@@ -46,7 +46,10 @@ protected function getMatomoUrl()
         }
 
         // this is a workaround when force_ssl=1, and the HTTPS URL is not fetchable from CLI
-        $url = str_replace('https://localhost', 'http://localhost', $url);
+        $parsed_url = parse_url($url);
+        if($parsed_url['scheme'] == 'https'){
+            $url = str_replace('https://', 'http://', $url);
+        }
```

Comment:
Wouldn't a `$url = str_replace('https://', 'http://', $url);` do exactly the same? 🤔 

But I guess replacing that always might make problems if a site is only available through `https`

Reply:
Why replace something if it is not used? Practice should be too look for if it is needed to replace before replacing anything. 
Yes if someone is running this plugin on a site that is only available via https it would be bad, and I would rather have a system setting for this - so the plugin knows what protocol to use. That I thought of fixing later on.